### ISSUE TITLE
Display associated sub-streams when a stream is selected.

### ISSUE THREAD

**Issue by plugyawn** [zulip/zulip-terminal#1190]
The web-app displays sub-streams when the stream is selected, which is a convenient way to narrow to them when needed. 

For reference, the implementation should look something like this, from https://github.com/Bios-Marcel/cordless, an open-source Discord terminal client. 
<img width="195" alt="Screenshot 2022-04-08 at 1 35 24 AM" src="https://user-images.githubusercontent.com/76529011/162287132-c920f57a-8083-4b47-8411-63828ebd8af6.png">

**Comment by neiljp** [comment]
@plugyawn Are you aware of using the `t` key on a stream to show a list of topics? If not, I'd recommend reviewing the (user) tutorial.

In terms of other directions I agree we could show a nested view a little like the web app, though even there it can be toggled into an all-topics view, like we achieve with `t` straight away.

There was talk of using modifier keys to auto-reset views, eg. after selecting something to narrow to, but I don't remember if it applied to `t`.

We could also improve UI hints on the screen that `t` exists, other than in the help menu, which we've improved over time for other areas. We perhaps considered having this at the base of the stream/topic list in a topic, you may be able to find it through searching.

**Comment by plugyawn** [comment]
@neiljp Sorry, yeah, I forgot that it existed; I've read the user tutorial, must've slipped my mind. 

But I think the issue still stands -- once you press `enter` to narrow to a stream, the indication that you've narrowed to a particular stream needs to be more explicit. 

So when you're in the all-messages view, the streams are not unravelled, but when you narrow to, say, zulip-terminal, the last 4 or 5 topics are listed as a tree, which serves both as a simple way to show, 
- the stream you've narrowed to (since the header is not a great way to show the stream that's open -- you're looking at the left column when selecting the stream and then looking at the header to confirm if it's the right stream that opened.
That provides a feedback for the stream selection.
- Gives users an easy way to jump between topics, without having to go through the hassle of knowing the `t` command and actually using it to navigate. Also, both `t` and `esc` should allow users to get back to the streams page, right now it's just `t`.

Also, totally think we should hint towards the `t` functionality more.

**Comment by neiljp** [comment]
@plugyawn Since this is generating more discussion than focus on a single point, and to generate wider thoughts, I've put some of my responses into a topic in the stream provisionally titled `Panel-follows-narrow`.

I'd suggest we raise the other points separately, since they're items to improve without such a feature enabled, or at least which can be discussed/developed in parallel. Note though that `esc` historically (and currently) is equivalent to `a` (at least on a message), so that could also be confusing, though we do use other keys differently in different contexts.

**Comment by plugyawn** [comment]
Yep! Sounds like the best way to do this. Should I close this issue and raise individual issues for the discussed points? 

Also -- about the `esc` thing, thanks for the context! Makes more sense now.

**Comment by neiljp** [comment]
Let's leave this issue open. I'm going to rename the issue slightly to represent what seems to be the main point in the context of ZT.
